bot/cogs/team.py

```diff
@@ -0,0 +1,346 @@
+
+    @team.command(
+        help="Move all members in a team channel to given voice channel using: !team gather [Voice channel]"
+    )
+    @commands.has_permissions(administrator=True)
+    async def gather(self, ctx, vc: VoiceChannel):
+        for team in self.teams:
+            for member in team.vc.members:
+                await member.move_to(vc)
+        await reply_and_delete(ctx, "Successfully moved members")
+
+    @team.group(
+        name="divide",
+        invoke_without_command=True,
+        help="Get help for team divide commands",
+    )
+    async def divide(self, ctx):
+        await reply_and_delete(ctx, "'team divide' needs a sub-command!")
+        await ctx.send_help(self.team.divide)
+
+    @divide.command(
+        help="Split current voice channel into number groups using: !team divide random [number of groups]"
+    )
+    @commands.has_permissions(administrator=True)
+    async def random(self, ctx, n: str):
+        # handle parameter
+        try:
+            n = int(n)
+        except:
+            await reply_and_delete(ctx, "Number of groups should be an integer value.")
+            return
+
+        vc = ctx.author.voice.channel
+        if not vc:
+            await reply_and_delete(
+                ctx, "You have to be connected to a voice channel to use this command."
+            )
+            return
+
+        connected = [m for m in vc.members if not m.bot]
+        if len(connected) < n:
+            await reply_and_delete(
+                ctx,
+                "There are less than "
+                + str(n)
+                + " users connected to your voice channel.",
+            )
+            return
+
+        shuffle(connected)
+        divided_teams = [
+            [connected[i] for i in range(len(connected)) if (i % n) == r]
+            for r in range(n)
+        ]
+
+        i = 0
+        for t in divided_teams:
+            self.create_team(t, "Team " + str(i))
+            i += 1
+
+    @divide.command(
+        help="Split current voice channel into teams based on their role: !team divide role"
+    )
+    @commands.has_permissions(administrator=True)
+    async def role(self, ctx, *args):
+        vc = ctx.author.voice.channel
+        if not vc:
+            await reply_and_delete(
+                ctx, "You have to be connected to a voice channel to use this command."
+            )
+            return
+        connected = [m for m in vc.members if not m.bot]
+        roles = [await string_to_role(ctx, arg) for arg in args]
+
+        divided_teams = [[] for _ in range(len(roles))]
+        for c in connected:
+            placed = False
+            for i in range(len(roles)):
+                if roles[i] in c.roles and not placed:
+                    divided_teams[i].append(c)
+                    placed = True
+
+        i = 0
+        for t in divided_teams:
+            self.create_team(t, roles[i].name)
+            i += 1
+
+    @tasks.loop(minutes=1)
```

Comment:
This interval might be a bit high, but on the other hand, the operation is not that intense.

Reply:
5 minutes would also be fine, then it is possible for people to return. Or maybe 10